### PR TITLE
AITOOL-2791: Update demo-fcm repo to use SICAP 1.0.0 version

### PR DESCRIPTION
**Pull request by DionisioRD**
# [AITOOL-2791](https://lampkicking.atlassian.net/browse/AITOOL-2791)

## What?
Bump the version of the FCM to the new SICAP beta release. Also added some handlers to request the optional parameters of the AI services API `Level of assurance` and `Metadata device`

## Why?
To update the references on how to integrate the FCM using the Yoti API to call AI services.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -87,20 +123,39 @@ const App = () => {
...
-          <div className={classes.faceCapture}>
-            <FaceCapture
-              captureMethod={CAPTURE_METHOD.AUTO}
-              onSuccess={onSuccess}
-              onError={onError}
-            />
+          <div className={classes.scanContainer}>
+            <div className={classes.faceCaptureWrapper}>
+              <FaceCapture
+                captureMethod={CAPTURE_METHOD.AUTO}
```
Comment (MiguelJS) [inline src/App.js:129]: I think AUTO is the default mode we could remove it, no?

File: server.js [hunk 2]
```diff
@@ -10,15 +10,26 @@ app.use(bodyParser.urlencoded({ extended: true }));
...
 
   const request = new RequestBuilder()
     .withBaseUrl(process.env.BASE_URL)
     .withPemFilePath(process.env.PEM_FILE_PATH)
     .withEndpoint(process.env.ENDPOINT)
-    .withPayload(new Payload({ img }))
+    .withPayload(new Payload(data))
     .withMethod("POST")
     .withHeader("X-Yoti-Auth-Id", process.env.SDK_ID)
+    .withQueryParam("secure", !!secure)
```
Comment (MiguelJS) [inline server.js:32]: It will allow us to call:
- `?secure=true`
- `?secure=false`
But not the endpoint without "secure" in the query, no?

Comment (DionisioRD) [reply server.js:32]: Correct :+1:

File: src/App.js (hunk 1 above)
Comment (DionisioRD) [reply src/App.js:129]: You are right, didn't thought about that :)

File: src/App.js [hunk 3]
```diff
@@ -55,15 +67,19 @@ const useStyles = makeStyles((theme) =>
...
   const classes = useStyles();
 
-  const onSuccess = ({ image }) => {
-    setImage(image);
+  const onSuccess = ({ img, secure }) => {
+    setImage(img);
+    const fcmResponse = { img, secure };
     service
-      .predict(image)
+      .predict(fcmResponse, levelOfAssurance, metadataDevice)
```
Comment (tonygcs) [inline src/App.js:82]: I think with this change you can avoid some object mappings in `server.js` and `src/api/api.js` files.
```suggestion
      .predict({ 
        ...fcmResponse,
        level_of_assurance: levelOfAssurance ? levelOfAssurance : undefined,
        metadata: metadataDevice ? { device: metadataDevice } : undefined,
      })
```
What do you think?

File: src/App.js [hunk 4]
```diff
@@ -77,6 +93,26 @@ const App = () => {
...
+
+  const onLevelOfAssuranceChange = (event) => {
+    if (event.target.value) {
+      setLevelOfAssurance(
+        event.target.value !== levelOfAssurance ? event.target.value : ""
+      );
+    }
+  };
+
+  const metadataDevices = ["laptop", "mobile", "unknown"];
```
Comment (tonygcs) [inline src/App.js:106]: Move this outside the component, please. I also wonder if the user should select this option or we should calculate it as this demo is like a real application the integrators can use as example.

File: server.js (hunk 2 above)
Comment (MiguelJS) [reply server.js:32]: Ok, if we want to test without the query parameter we just need to delete the line
     .withQueryParam("secure", !!secure)
and recompile the package no?

File: src/App.js (hunk 4 above)
Comment (MiguelJS) [reply src/App.js:106]: At the moment, it is the integrator who decides the metadata and therefore the antispoofing they want to use in the request for our services.
If we make it mandatory to read the METADATA and use it for SICAP secure version. We could enforce it in the future, wdyt?

**Comment by MiguelJS** [comment]
@DionisioRD could you update the README file by adding the new requests options, and adding information about version 1.0.0-beta.2 changes?

File: server.js (hunk 2 above)
Comment (DionisioRD) [reply server.js:32]: Yes, that would be possible.

File: src/App.js (hunk 1 above)
Comment (DionisioRD) [reply src/App.js]: Done

File: src/App.js (hunk 3 above)
Comment (DionisioRD) [reply src/App.js]: You both were right :+1:

File: src/App.js (hunk 4 above)
Comment (DionisioRD) [reply src/App.js]: Metadata has been removed. I moved the LevelsOfAssurance out of the component tho.

**Comment by DionisioRD** [comment]
I updated to 1.0.0-beta.3 the FCM as there were no breaking changes.

**Review by tonygcs — APPROVED** [review]

**Review by pegondo — APPROVED** [review]

**Pull request merged**